pkg/engine/engine.go

```diff
@@ -426,6 +427,40 @@ func (e *Engine) Kill(id string) error {
 	return nil
 }
 
+// CleanUpTasks deletes all tasks in the store that are currently being processed
+func (e *Engine) CleanUpTasks() error {
+
+	allTasks, err := e.store.AllTasks()
+	if err != nil {
+		return err
+	}
+
+	deleted := 0
+	for _, aTask := range allTasks {
+		if aTask.State().State == task.StateProcessing {
+			// If the engine has a signal for the task, that means it's being tracked actively
+			if e.hasSignal(aTask.ID) {
+				logging.S().Debugw("Skipping cleanup of task, as it is still active", "task", aTask.ID, "state", aTask.State())
+				continue
+			}
+
+			logging.S().Infow("Cleaning up inactive task", "task", aTask.ID, "type", aTask.Name())
+			err = e.store.Delete(aTask.ID)
```

Comment:
Sorry for not replying earlier, I haven't had time to look into the storage part.

https://github.com/testground/testground/issues/1293 is talking about,
> containers lingering forever as completed in Kubernetes.

And the task is
> cleans up the Kubernetes namespace from in-progress / crashed / completed testplan runs on startup

But it looks like this PR is just removing tasks logs from Testground's leveldb database.
We don't want to remove logs from the history, we want to clean lingering resources.

This is the sort of code I expect to see:
https://github.com/testground/testground/blob/3400b0e10b58ba772f3a1a1c37f2c73fbc0e1972/pkg/runner/cluster_k8s.go#L346-L357

- Could you make sure we're not removing any tasks, but at least marking them as cancelled?
- Could you confirm this PR is cleaning up the Kubernetes namespace from in-progress / crashed / completed testplan on startup?

Reply:
Thanks for the explanation! I think you're right - I'll take some time to look through the code and test a solution on the new cluster.

Since it's been a while, I seem to remember there being a reason why I removed them from the DB - I'll see if I can dig it up.